src/components/apps/listing/TableView.js

```diff
@@ -207,6 +218,14 @@ function TableView(props) {
                                                 )}
                                                 isDisabled={app.disabled}
                                                 name={app.name}
+                                                onAppNameClicked={() =>
+                                                    router.push(
+                                                        getAppLaunchPath(
+                                                            app.system_id,
+                                                            app.id
+                                                        )
+                                                    )
+                                                }
```

Comment:
I added the `router` call here, since there's already a precedence for calling `router.push` in the analyses listing component, but I'm not sure if this is the right place for this call. Note that this will throw an exception in the listing stories for apps and analyses, since `router` is undefined there.

The alternative might be passing down a callback from the page component which can call `router.push` (similar to the new `launch` page in this PR).

Reply:
I guess passing callback from the page should work. I can refactor analyses listing as well.

Reply:
I'm also for having the pages define the routing.  At least in the case for Data, it was necessary to make the component reusable not within a page (as in the data selection drawer).